app/controllers/concerns/authorizable.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Authorizable
+  extend ActiveSupport::Concern
+
+  # Unless the request format is explicitly json Rails will mitigate the responsibility to CSR to handle it.
+  def ensure_valid_request
+    return render 'components/index' if !Rails.env.development? && !valid_api_request?
+  end
+
+  # Ensures that the user is logged in
+  def ensure_authenticated
```

Comment:
I think we need `ensure_unathenticated` either.

Reply:
Will do it in another PR